libs/wingsdk/src/shared-gcp/bucket.inflight.ts

```diff
@@ -53,20 +55,34 @@ export class BucketClient implements IBucketClient {
     }
   }
 
-  public async put(key: string, body: string): Promise<void> {
-    try {
-      await this.bucket.file(key).save(body);
-    } catch (error) {
-      throw new Error(`Failed to put object. (key=${key})`);
-    }
-  }
-
+  /**
+   * Put object into bucket with given body contents
+   *
+   * @param key Key of the object
+   * @param body string contents of the object
+   */
+  public async put(
+    key: string,
+    body: string,
+    opts?: BucketPutOptions
+  ): Promise<void> {
+    const options = {
+      contentType:
+        (opts?.contentType ?? mime.lookup(key)) || "application/octet-stream",
+    };
+    await this.bucket.file(key).save(body, options);
+  }
+
+  /**
+   * Put Json object into bucket with given body contents
+   *
+   * @param key Key of the object
+   * @param body Json object
+   */
   public async putJson(key: string, body: Json): Promise<void> {
-    try {
-      await this.put(key, JSON.stringify(body, null, 2));
-    } catch (error) {
-      throw new Error(`Failed to put JSON object. (key=${key})`);
-    }
+    await this.put(key, JSON.stringify(body, null, 2), {
```

Comment:
Why are we removing the try/catch logic?

Reply:
My rationale was probably to make it consistent with the [AWS implementation](https://github.com/winglang/wing/blob/693ee4b3a995a9c305479b32c10e0a87013ff125/libs/wingsdk/src/shared-aws/bucket.inflight.ts#L77). I believe error handling is not necessary here because the operation involves either creating a new object or overwriting an existing one.

Reply:
To make it consistent, we might consider introducing methods like `tryPut()` and `tryPutJson()`, akin to `tryGet()` and `tryGetJson()`, for situations where we need to manage failures directly.

Reply:
For the sake of consistency, we might consider introducing methods like `tryPut()` and `tryPutJson()`, akin to `tryGet()` and `tryGetJson()`, for situations where we need to manage failures directly.

Reply:
Got it - yeah this change makes sense to me. Unifying error messages to be more similar between targets could be a nice project in the future but the current code looks like it's discarding the thrown `error` which is not great for debugging, so I'm okay with just letting any errors bubble up